aten/src/ATen/native/cpu/ReduceOpsKernel.cpp

```diff
@@ -237,140 +237,6 @@ static void prod_kernel_impl(TensorIterator& iter) {
   }
 }
 
-template <typename scalar_t, typename acc_t>
-inline void norm_two_reduce_step(Vectorized<acc_t>& acc_vec, Vectorized<scalar_t>& data_vec) {
-  acc_vec += data_vec * data_vec;
-}
-
-template <>
-inline void norm_two_reduce_step(Vectorized<float>& acc_fvec, Vectorized<BFloat16>& data_bvec) {
-  Vectorized<float> data_fvec0, data_fvec1;
-  std::tie(data_fvec0, data_fvec1) = convert_bfloat16_float(data_bvec);
-  acc_fvec += data_fvec0 * data_fvec0;
-  acc_fvec += data_fvec1 * data_fvec1;
-}
-
-static void norm_kernel_tensor_iterator_impl(
-    TensorIterator& iter,
-    const Scalar& p) {
-  // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
-  float val;
-  if (p.isIntegral(false)) {
-    val = p.to<int64_t>();
-  } else if (p.isFloatingPoint()) {
-    // NOLINTNEXTLINE(cppcoreguidelines-narrowing-conversions,bugprone-narrowing-conversions)
-    val = p.to<double>();
-  } else {
-    AT_ERROR("norm_kernel_tensor_iterator_impl expects norm to be integer or float");
-  }
-  if (iter.numel() == 0) {
-    iter.output().fill_((val < 0) ? INFINITY : 0);
-    return;
-  }
-
-  // In the dispatch code blocks below, reduction kernels accumulate results as
-  // the type `acc_t`. When `scalar_t` is complex, `acc_t` is the downgraded
-  // real number type. Otherwise, `acc_t` and `scalar_t` are the same type.
-  if (val == 0) {
-    AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(kHalf, kBFloat16, iter.input_dtype(), "norm_cpu", [&] {
-      using acc_t = typename scalar_value_type<scalar_t>::type;
-      binary_kernel_reduce(
-        iter,
-        NormZeroOps<scalar_t, acc_t>(),
-        acc_t(0)
-      );
-    });
-  } else if (val == 1) {
-    AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(kHalf, kBFloat16, iter.input_dtype(), "norm_cpu", [&] {
-      using acc_t = typename scalar_value_type<scalar_t>::type;
-      binary_kernel_reduce(
-        iter,
-        NormOneOps<scalar_t, acc_t>(),
-        acc_t(0)
-      );
-    });
-  } else if (val == 2) {
-    // If we can vectorize over the last dimension and the dtype
-    // of the output is the same as that of the input,
-    // then we go through the vectorised path.
-    if (is_reduce_lastdim(iter) &&
```

Comment:
What happened to this code?

Reply:
It's up there in the new file. The only thing is that the comment is gone for no apparent reason

Reply:
Ah, I was expecting to see it next to `NormTwoOps` but they've been split apart.